.travis.yml

```diff
@@ -45,6 +53,14 @@ matrix:
                   latest_file="$(basename $latest_file .rootfs)"
                   # Find all files with that prefix and copy them to our cache dir
                   sudo find /var/snap/lxd/common/lxd/images/ -name $latest_file* -print -exec cp {} lxd_images/ \;
+              - |
+                  # If a schroot exists (i.e. we didn't fail before its
+                  # creation), tar it up (to preserve ownership/permissions)
+                  # and move it into the cached dir; no need to compress it
+                  # because Travis will do that anyway
+                  if [ -e /var/lib/schroot/chroots/xenial-amd64 ]; then
+                    sudo tar --sparse --xattrs --xattrs-include=* cf chroots/xenial-amd64.tar -C /var/lib/schroot/chroots/xenial-amd64 .
```

Comment:
Is $CWD /var/lib/schroot ?
It might be best to just use full path to files everywhere incase $CWD changes

```
schroot_dir=/var/lib/schroot/chroots/xenial-amd64
schroot_tar=${schroot_dir}.tar
if [ -e "${schroot_dir}" ]; then
   sudo tar .... -cf ${schroot_tar} -C ${schroot_dir}
fi
```

Reply:
Nope, `$CWD` is the directory which Travis runs the tests from (aka `$TRAVIS_BUILD_DIR`). The cache paths are relative to that, so we cache `$TRAVIS_BUILD_DIR/chroots`, which is where we're creating the tar. I'll see if I can use `$TRAVIS_BUILD_DIR` to make this a little more robust.

(We can't use an absolute path to directly cache the chroots because the unprivileged Travis user needs full read/write permissions on the cache directories.)

Reply:
OK, this is done. I'll submit a separate PR to change the LXD image caching to also use the environment variable, so we can start benefiting from this sooner rather than later.